src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java

```diff
@@ -94,14 +95,14 @@ public void run() {
 
   @Override
   public void onServiceFound(NsdServiceInfo service) {
-    if (targetServiceInfo.getServiceName().equals(service.getServiceName())) {
+    if (discoveredServiceInfo == null
+        && targetServiceInfo.getServiceName().equals(service.getServiceName())) {
       Log.d(TAG, "onServiceFound: found target service " + service);
 
       if (stopDiscoveryRunnable.cancel(false)) {
         nsdManager.stopServiceDiscovery(this);
       }
-
-      resolveService(service, callbackHandle, contextHandle, chipMdnsCallback);
+      discoveredServiceInfo = service;
     } else {
       Log.d(TAG, "onServiceFound: found service not a target for resolution, ignoring " + service);
```

Comment:
this log is not correct anymore. we should report duplicates.

Reply:
After the discussion below is completed, we will revise it together according to the revised contents.